tests/res/mocks/libs/bash_script.py

```diff
@@ -0,0 +1,57 @@
+import base64
+from pathlib import Path
+from typing import List, Optional
+
+__FILE_NAME_AND_CONTENT_TEMPLATE = """
+OUTPUT_FILE_NAME_{index}="{file_name}"
+OUTPUT_FILE_CONTENT_{index}="{file_content}"
+sleep 1
+printf '%s' "$OUTPUT_FILE_CONTENT_{index}" | base64 -d > $OUTPUT_FILE_NAME_{index}
+"""
+
+__STDOUT_TEMPLATE = """
+STDOUT_FILE_CONTENT="{stdout}"
+sleep 1
+printf '%s' "$STDOUT_FILE_CONTENT" | base64 -d >&1
+"""
+
+__STDERR_TEMPLATE = """
+STDERR_FILE_CONTENT="{stderr}"
+sleep 1
+printf '%s' "$STDERR_FILE_CONTENT" | base64 -d >&2
+"""
+
+
+def encode_single_file(index: int, file_name: str, file_content: bytes) -> str:
```

Comment:
What is index parameter ? Do you mean a line number ? If yes, then call it `line_no`

Reply:
Index is used to template, e.g we have two files: `example001_fort.21` and `example001_fort.22`.
Because, this is generic script, name and content of these files will be stored in bash script as variables:

```bash
OUTPUT_FILE_NAME_0="example001_fort.21"
OUTPUT_FILE_CONTENT_0="<base64_encoded_content>"

OUTPUT_FILE_NAME_1="example001_fort.21"
OUTPUT_FILE_CONTENT_1="<base64_encoded_content>"
```

I can refactor this script to use class. Then, index will be stored in its instance.

Reply:
Use `file_no` instead